tools/shims/src/applicationinsights-shims.ts

```diff
@@ -0,0 +1,111 @@
+ * - globalThis (New standard)
+ * - self (Will return the current window instance for supported browsers)
+ * - window (fallback for older browser implementations)
+ * - global (NodeJS standard)
+ * - <null> (When all else fails)
+ * While the return type is a Window for the normal case, not all environments will support all
+ * of the properties or functions.
+ */
+export function getGlobal(): Window {
+    if (typeof globalThis !== strShimUndefined && globalThis) {
+        return globalThis;
+    }
+
+    if (typeof self !== strShimUndefined && self) {
+        return self;
+    }
+
+    if (typeof window !== strShimUndefined && window) {
+        return window;
+    }
+
+    if (typeof global !== strShimUndefined && global) {
+        return global;
+    }
+
+    return null;
+}
+
+/**
+ * Creates an object that has the specified prototype, and that optionally contains specified properties. This helper exists to avoid adding a polyfil
+ * for older browsers that do not define Object.create eg. ES3 only, IE8 just in case any page checks for presence/absence of the prototype implementation.
+ * Note: For consistency this will not use the Object.create implementation if it exists as this would cause a testing requirement to test with and without the implementations
+ * @param obj Object to use as a prototype. May be null
+ */
+export function objCreateFn(obj: any): any {
+    var func = Object["create"];
+    // Use build in Object.create
+    if (func) {
+        // Use Object create method if it exists
+        return func(obj); 
+    }
+    if (obj == null) { 
+        return {}; 
+    }
+    var type = typeof obj;
+    if (type !== strShimObject && type !== strShimFunction) { 
+        throw new TypeError('Object prototype may only be an Object:' + obj); 
+    }
+
+    function tmpFunc() {}
+    tmpFunc[strShimPrototype] = obj;
+
+    return new (tmpFunc as any)();    
+}
+
+export function __assignFn(t: any) {
+    for (var s, i = 1, n = arguments.length; i < n; i++) {
+        s = arguments[i];
+        // tslint:disable-next-line: curly
+        for (var p in s) if (Object[strShimPrototype][strShimHasOwnProperty].call(s, p)) t[p] = s[p];
+    }
+    return t;
+}
+
+var __extendStaticsFn = function(d: any, b: any): any {
+    __extendStaticsFn = Object["setPrototypeOf"] ||
+        ({ __proto__: [] } instanceof Array && function (d: any, b: any) { d.__proto__ = b; }) ||
+        // tslint:disable-next-line: curly
+        function (d: any, b: any) { for (var p in b) if (b[strShimHasOwnProperty](p)) d[p] = b[p]; };
+    return __extendStaticsFn(d, b);
+};
+
+export function __extendsFn(d: any, b: any) {
+    __extendStaticsFn(d, b);
+    function __() { this.constructor = d; }
+    // tslint:disable-next-line: ban-comma-operator
+    d[strShimPrototype] = b === null ? objCreateFn(b) : (__[strShimPrototype] = b[strShimPrototype], new (__ as any)());
+}
+
+let globalObj:any = getGlobal() || {};
```

Comment:
From here on is the primary "tricks" to wire up to the global __extends; __assign and local versions (lines 109, 110).
The primary change that this introduces (as opposed to not using tslib), is that moving forward *if* the consumer (of the npm packages) has defined their own __extends() and/or __assign() then this code WILL use them, previously the tslib versions where the only possible usages because tslib was a injected (via require) and used as tslib_1.__extend() rather than the global versions.